### PR TITLE
Enable preview scripts

### PR DESCRIPTION
**Pull request by xivira**
Hello!

I have made some modifications to allow the generation of preview PDFs:

1. Added `build.sh` and `parser.py` that provide better feedback on what's going on during the build process.
2. Added a `--preview` option to `app.go` that applies a unique-module filter over the generated module array.
3. Added a `PREVIEW` environment variable for the `build.sh` and `single.sh` scripts

I will use these to generate previews on the website!

### CODE REVIEW

**Comment by xivira** [comment]
Hello!

I have made some modifications to allow the generation of preview PDFs:

1. Added `build.sh` and `parser.py` that provide better feedback on what's going on during the build process.
2. Added a `--preview` option to `app.go` that applies a unique-module filter over the generated module array.
3. Added a `PREVIEW` environment variable for the `build.sh` and `single.sh` scripts

I will use these to generate previews on the website!

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+go.sum
```
Comment (kudrykv) [inline .gitignore:1]: The `go.sum` should not be ignored.

File: app/app.go [hunk 2]
```diff
@@ -132,3 +142,28 @@ var ComposerMap = map[string]Composer{
...
+
+	return func(index int, element page.Module) bool {
+		if _, present := found[element.Tpl]; present {
+			return false
+		} else {
+			found[element.Tpl] = true
+			return true
+		}
+	}
+}
```
Comment (kudrykv) [inline app/app.go:169]: This func is called once directly. Hence, the closure for `found` is redundant.

This construct as a whole looks heavy. What other filters do you anticipate?

File: build.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+# If no target year is passed in though argv[0], use next year as default value
+if [ $# -eq 1 ]; then
+    TARGET_YEAR=$1
+else
+    TARGET_YEAR=$(expr $(date +"%Y") + 1)
+fi
+
+CONFIG_FILES='cfg/base.yaml,cfg/template_months_on_side.yaml,cfg/sn_a5x.mos.default.yaml,cfg/sn_a5x.mos.default.dailycal.yaml'
```
Comment (kudrykv) [inline build.sh:10]: Should it be configurable?

File: preview.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+# If no target year is passed in though argv[0], use next year as default value
+if [ $# -eq 1 ]; then
+    TARGET_YEAR=$1
+else
+    TARGET_YEAR=$(expr $(date +"%Y") + 1)
+fi
+
+CONFIG_FILES='cfg/base.yaml,cfg/template_months_on_side.yaml,cfg/sn_a5x.mos.default.yaml,cfg/sn_a5x.mos.default.dailycal.yaml'
```
Comment (kudrykv) [inline preview.sh:10]: Should this be configurable?

File: .gitignore (hunk 1 above)
Comment (xivira) [reply .gitignore]: My bad, I just started learning Go and thought it was something similar to Node's `package-lock.json`. Issue in the following commit:

https://github.com/kudrykv/latex-yearly-planner/pull/27/commits/abe2000686ad75946a3cf085b202573b70644e98

File: app/app.go (hunk 2 above)
Comment (xivira) [reply app/app.go]: Removed redundant function in the following commit: https://github.com/kudrykv/latex-yearly-planner/pull/27/commits/6bfe6909f5cc94207914405db31ae5828de13d1c.

**Comment by kudrykv** [comment]
Please remove `plannergen` binary. Then it should be ready to go.

Be also aware though, that I'm currently working on yet another refactor. It'll take a while to implement, but generally, the interaction with CLI will change a little. We'll need to sync up again then.

**Comment by kudrykv** [comment]
Please remove `plannergen` binary. Then it should be ready to go.

Be also aware though, that I'm currently working on yet another refactor. It'll take a while to implement, but generally, the interaction with CLI will change a little. We'll need to sync up again then.

**Comment by xivira** [comment]
> [...]
> Be also aware though, that I'm currently working on yet another refactor. It'll take a while to implement, but generally, the interaction with CLI will change a little. We'll need to sync up again then.

Absolutely, I've been very busy as of late but I'm still very interested in the project. Looking forward to the refactor!

**Review by kudrykv — APPROVED** [review]

**Pull request merged**